wifiphisher/extensions/handshakeverify.py

```diff
@@ -1,370 +1,80 @@
 # pylint: skip-file
-# FIXME Pylint complains about EAPOL.
 """
-Extension that capture the four way handshake and
-do the verification whether the password given by
--pK option is valid
+Extension that verifies WPA key by precaptured handshake using cowpatty
 """
+import subprocess
+from collections import defaultdict
+import shlex
 
-import binascii
-import hashlib
-import hmac
-import logging
-from collections import defaultdict, deque
-
-import scapy.layers.dot11 as dot11
-import wifiphisher.common.constants as constants
 import wifiphisher.common.extensions as extensions
-from pbkdf2 import PBKDF2
-from scapy.all import rdpcap
-
-logger = logging.getLogger(__name__)
-
-# define the verification state
-DONE, FAIL, NOT_YET = list(range(3))
-
-# backward compatible for scapy EAPOL
-try:
-    EAPOL = dot11.EAPOL
-except AttributeError:
-    # incase scapy version >= 2.4.0
-    import scapy.layers.eap as eap
-    EAPOL = eap.EAPOL
-
 
-def is_valid_handshake_capture(handshake_path):
-    """
-    Check if valid handshake capture is found
-    :param handshake_path: file path of handshake
-    :type handshake_path: str
-    :return: None
-    :rtype: None
-    """
-    pkts = rdpcap(handshake_path)
-    eapols = []
-    # get all the KEY type EAPOLs
-    for pkt in pkts:
-        # pkt is Dot11 and is not retried frame
-        if pkt.haslayer(dot11.Dot11) and not pkt.FCfield & (1 << 3):
-            # pkt is EAPOL and KEY type
-            if pkt.haslayer(EAPOL) and pkt[EAPOL].type == 3:
-                eapols.append(pkt)
-
-    num_of_frames = len(eapols)
-    for index in range(num_of_frames):
-        if num_of_frames - index > 3:
-            ap_bssid = eapols[index].addr2
-            # from AP to STA
-            msg1 = eapols[index]
-            # from STA to AP
-            msg2 = eapols[index + 1]
-            # from AP to STA
-            msg3 = eapols[index + 2]
-            # from STA to AP
-            msg4 = eapols[index + 3]
-
-            if msg1.addr2 == ap_bssid and\
-                    msg3.addr2 == ap_bssid and\
-                    msg2.addr1 == ap_bssid and\
-                    msg4.addr1 == ap_bssid:
-                logger.info("Get valid handshake frames")
-                return True
-        else:
+def get_process_result(command_string):
+    command = shlex.split(command_string)
+    process = subprocess.Popen(command,
+        stdout=subprocess.PIPE,
+        universal_newlines=True)
+    output = ""
+    while True:
+        output += process.stdout.readline().strip()
+        code = process.poll()
+        if code is not None:
+            for lines in process.stdout.readlines():
+                output += lines.strip()
             break
-    logger.info("No valid handshake frames exists")
-    return False
+    return output
 
+def is_valid_handshake_capture(filename):
+    command = '/bin/cowpatty -c -r {}'.format(filename)
```

Comment:
Can we return an error if cowpatty is not installed?

Reply:
Ok, I willl accomplish it as soon as possible.